File: SDKs/amazonaws.com/ec2/2016-11-15/go/pkg/models/operations/postdescribeclientvpntargetnetworks.go
package operations

type PostDescribeClientVpnTargetNetworksActionEnum string

const (
	PostDescribeClientVpnTargetNetworksActionEnumDescribeClientVpnTargetNetworks PostDescribeClientVpnTargetNetworksActionEnum = "DescribeClientVpnTargetNetworks"
)

type PostDescribeClientVpnTargetNetworksVersionEnum string

const (
	PostDescribeClientVpnTargetNetworksVersionEnumTwoThousandAndSixteen1115 PostDescribeClientVpnTargetNetworksVersionEnum = "2016-11-15"
)

type PostDescribeClientVpnTargetNetworksQueryParams struct {
	Action     PostDescribeClientVpnTargetNetworksActionEnum  `queryParam:"style=form,explode=true,name=Action"`
	MaxResults *string                                        `queryParam:"style=form,explode=true,name=MaxResults"`
	NextToken  *string                                        `queryParam:"style=form,explode=true,name=NextToken"`
	Version    PostDescribeClientVpnTargetNetworksVersionEnum `queryParam:"style=form,explode=true,name=Version"`
}

type PostDescribeClientVpnTargetNetworksHeaders struct {
	XAmzAlgorithm     *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
}

type PostDescribeClientVpnTargetNetworksRequest struct {
	QueryParams PostDescribeClientVpnTargetNetworksQueryParams
	Headers     PostDescribeClientVpnTargetNetworksHeaders
	Request     []byte `request:"mediaType=text/xml"`
}

type PostDescribeClientVpnTargetNetworksResponse struct {
	Body        []byte
	ContentType string
	StatusCode  int64
}

File: SDKs/amazonaws.com/ec2/2016-11-15/typescript/src/sdk/models/operations/postdescribeclientvpntargetnetworks.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostDescribeClientVpnTargetNetworksActionEnum {
    DescribeClientVpnTargetNetworks = "DescribeClientVpnTargetNetworks"
}

export enum PostDescribeClientVpnTargetNetworksVersionEnum {
    TwoThousandAndSixteen1115 = "2016-11-15"
}


export class PostDescribeClientVpnTargetNetworksQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: PostDescribeClientVpnTargetNetworksActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=MaxResults" })
  maxResults?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=NextToken" })
  nextToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: PostDescribeClientVpnTargetNetworksVersionEnum;
}


export class PostDescribeClientVpnTargetNetworksHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class PostDescribeClientVpnTargetNetworksRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostDescribeClientVpnTargetNetworksQueryParams;

  @SpeakeasyMetadata()
  headers: PostDescribeClientVpnTargetNetworksHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=text/xml" })
  request?: Uint8Array;
}


export class PostDescribeClientVpnTargetNetworksResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
